plugins/PostProcessingPlugin/scripts/PurgeLinesAndUnload.py

```diff
@@ -0,0 +1,881 @@
+        # Set the default E adjustment
+        self._instance.setProperty("adjust_e_loc_to", "value", -abs(round(float(self.extruder[0].getProperty("retraction_amount", "value")), 1)))
+
+    def getSettingDataString(self):
+        return """{
+            "name": "Purge Lines and Unload Filament",
+            "key": "PurgeLinesAndUnload",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+            "add_purge_lines":
+                {
+                    "label": "Add Purge Lines to StartUp",
+                    "description": "The purge lines can be left, right, front or back.  If there are purge lines present in the StartUp Gcode remove them or comment them out before using this script.  You don't want to double dip.",
+                    "type": "bool",
+                    "default_value": false,
+                    "value": false,
+                    "enabled": true
+                },
+                "purge_line_location":
+                {
+                    "label": "    Purge Line Location",
+                    "description": "What edge of the build plate should have the purge lines.  If the printer is 'Elliptical' then it is assumed to be an 'Origin At Center' printer and the purge lines are 90° arcs.",
+                    "type": "enum",
+                    "options": {
+                        "left": "On left edge (Xmin)",
+                        "right": "On right edge (Xmax)",
+                        "bottom": "On front edge (Ymin)",
+                        "top": "On back edge (Ymax)"},
+                    "default_value": "left",
+                    "enabled": "add_purge_lines"
+                },
+                "purge_line_length":
+                {
+                    "label": "    Purge Line Length",
+                    "description": "Select 'Full' for the entire Height or Width of the build plate.  Select 'Half' for shorter purge lines.  NOTE: This has no effect on elliptic beds.",
+                    "type": "enum",
+                    "options": {
+                        "purge_full": "Full",
+                        "purge_half": "Half"},
+                    "default_value": "purge_full",
+                    "enabled": "add_purge_lines and is_rectangular"
+                },
+                "move_to_start":
+                {
+                    "label": "Circle around to layer start",
+                    "description": "Depending on where the 'Layer Start X' and 'Layer Start Y' are for the print, the opening travel move can pass across the print area and leave a string there.  This option will generate an orthogonal path that moves the nozzle around the edges of the build plate and then comes in to the Start Point.  The nozzle will drop and touch the build plate at each stop in order to nail down the string so it doesn't follow in a straight line.",
```

Comment:
Touching the build plate with the hot nozzle can be destructive in some configurations. Can you make the warning a bit more visible ? Maybe add a :warning: emoji ? Not sure this is possible...